pyproj/_crs.pyx

```diff
@@ -117,15 +244,165 @@ cdef class AreaOfUse:
+        if self.projobj != NULL:
+            proj_destroy(self.projobj)
+        if self.projctx != NULL:
+            proj_context_destroy(self.projctx)
+
+    def _post_setup(self):
+        """
+        Setup to run after create
+        """
+        if self.projobj == NULL:
+            return
+        cdef const char* proj_name = proj_get_name(self.projobj)
+        self.name = decode_or_undefined(proj_name)
+        self.projctx = get_pyproj_context()
+
+    def to_wkt(self, version="WKT2_2018", pretty=False):
+        """
+        Convert the projection to a WKT string.
+
+        Version options:
+          - WKT2_2015
+          - WKT2_2015_SIMPLIFIED
+          - WKT2_2018
+          - WKT2_2018_SIMPLIFIED
+          - WKT1_GDAL
+          - WKT1_ESRI
+
+
+        Parameters
+        ----------
+        version: str
+            The version of the WKT output. Default is WKT2_2018.
+        pretty: bool
+            If True, it will set the output to be a multiline string. Defaults to False.
+ 
+        Returns
+        -------
+        str: The WKT string.
+        """
+        return _to_wkt(self.projctx, self.projobj, version, pretty=pretty)
+
+    def __str__(self):
+        return self.name
+
+    def __repr__(self):
+        return self.to_wkt(pretty=True)
+
+    def is_exact_same(self, Base other):
+        """Compares projections to see if they are exactly the same."""
+        return proj_is_equivalent_to(
+            self.projobj, other.projobj, PJ_COMP_STRICT) == 1
+
+    def __eq__(self, Base other):
+        """Compares projections to see if they are equivalent."""
+        return proj_is_equivalent_to(
+            self.projobj, other.projobj, PJ_COMP_EQUIVALENT) == 1
+
+
+_COORD_SYSTEM_TYPE_MAP = {
+    PJ_CS_TYPE_UNKNOWN: "unknown",
+    PJ_CS_TYPE_CARTESIAN: "cartesian",
+    PJ_CS_TYPE_ELLIPSOIDAL: "ellipsoidal",
+    PJ_CS_TYPE_VERTICAL: "vertical",
+    PJ_CS_TYPE_SPHERICAL: "spherical",
+    PJ_CS_TYPE_ORDINAL: "ordinal",
+    PJ_CS_TYPE_PARAMETRIC: "parametric",
+    PJ_CS_TYPE_DATETIMETEMPORAL: "datetimetemporal",
+    PJ_CS_TYPE_TEMPORALCOUNT: "temporalcount",
+    PJ_CS_TYPE_TEMPORALMEASURE: "temporalmeasure",
+}
+
+cdef class CoordinateSystem(Base):
+    """
+    Coordinate System for CRS
+
+    Attributes
+    ----------
+    name: str
+        The name of the coordinate system.
+
+    """
+    def __init__(self):
+        self._axis_list = None
+
+    @staticmethod
+    cdef create(PJ_CONTEXT* projcontext, PJ* projobj):
+        cdef CoordinateSystem coord_system = CoordinateSystem()
+        coord_system.projobj = proj_crs_get_coordinate_system(
+            projcontext,
+            projobj
+        )
+        if coord_system.projobj == NULL:
+            return None
+
+        cdef PJ_COORDINATE_SYSTEM_TYPE cs_type = proj_cs_get_type(
+            coord_system.projctx,
+            coord_system.projobj
+        )
+        coord_system.name = _COORD_SYSTEM_TYPE_MAP[cs_type]
```

Comment:
I found that the `type` matched the name in the `wkt` and the `proj_get_name` returned undefined. So, this made sense to me. Also, the `name` is used in the `__str__` method and I figured this would best represent the coordinate system name.

Reply:
> I found that the `type` matched the name in the `wkt` and the `proj_get_name` returned undefined. So, this made sense to me. Also, the `name` is used in the `__str__` method and I figured this would best represent the coordinate system name.

OK, that makes sense (the reason that I mentioned this, is that in general the `name` is kind of a (somewhat?) unique description of the object, while here you have many entries in the database of the same type. But, I think in general you don't care about those different 2D cartesian coordinate systems, so yes, it sounds good to treat that as the name.

Reply:
I will double check on this since the DB has other names, just to be sure.

Reply:
> I will double check on this since the DB has other names, just to be sure.

I think CoordinateSystem is one of the only tables that has no name column:

https://github.com/OSGeo/proj.4/blob/fc395adffc379ec2281288ee4d87a09eca9653bc/data/sql/proj_db_table_defs.sql#L131-L135

Reply:
That's unfortunate.

Reply:
I don't think it's really a problem. We can just either have the `name` as None or use the type of `name` in this case, as you did.